src/blob_file_set.cc

```diff
@@ -329,19 +329,36 @@ void BlobFileSet::GetObsoleteFiles(std::vector<std::string>* obsolete_files,
   obsolete_manifests_.clear();
 }
 
-void BlobFileSet::GetAllFiles(std::vector<std::string>* files, uint64_t* manifest_file_size) {
-  for (auto it = column_families_.begin(); it != column_families_.end(); ++it) {
-    auto& blob_storage = it->second;
-    // Get live blob file
-    blob_storage->GetAllFiles(files);
+void BlobFileSet::GetAllFiles(std::vector<std::string>* files, std::vector<VersionEdit>* edits) {
+  std::vector<std::string> all_blob_files;
+  
+  edits->clear();
+  edits->reserve(column_families_.size());
+  
+  for (auto& cf : column_families_) {
+    VersionEdit edit;
+    auto& blob_storage = cf.second;
+    // Get all blob files
+    blob_storage->GetAllFiles(&all_blob_files);
+    // Add all blob files to version_edit
+    edit.SetColumnFamilyID(cf.first);
+    edit.SetNextFileNumber(next_file_number_.load());
+    for (auto& file : blob_storage->files_) {
+      edit.AddBlobFile(file.second);
+    }
+    edits->emplace_back(edit);
+  }
+
+  files->clear();
+  files->reserve(all_blob_files.size() + 2);
+  
+  for (auto& live_file : all_blob_files) {
+    files->emplace_back(live_file);
   }
-  // Append current MANIFEST and CURRENT file name
-  std::string file_name = DescriptorFileName("/titandb", manifest_file_number_);
-  files->push_back(file_name);
-  files->push_back(CurrentFileName("/titandb"));
 
-  // Find current length of titandb manifest file while holding the mutex lock
-  *manifest_file_size = manifest_->file()->GetFileSize();
+  // Append current MANIFEST and CURRENT file name
+  files->emplace_back(DescriptorFileName("/titandb", manifest_file_number_));
```

Comment:
the origin manifest file should be excluded?

Reply:
Oh, that's true

Reply:
No, we only obtain the name of the file, we use the old `manifest_file_number_`, but the content is produced by ourselves. If we excluded it, we will lose the manifest file name.